src/middlewared/middlewared/plugins/security/update.py

```diff
@@ -48,8 +50,68 @@ async def configure_fips_on_ha(self, is_ha, job):
             except Exception:
                 # something extravagant happened, so we'll just play it safe and say that
                 # another reboot is required
-                await self.middleware.call('failover.reboot.add_remote_reason', RebootReason.FIPS.name,
-                                           RebootReason.FIPS.value)
+                await self.middleware.call('failover.reboot.add_remote_reason', reason.name,
+                                           reason.value)
+
+    @private
+    async def configure_stig(self, data=None):
+        if data is None:
+            data = await self.config()
+
+        if not data['enable_gpos_stig']:
+            await self.middleware.call('auth.set_authenticator_assurance_level', 'LEVEL_1')
+            return
+
+        # Per security team STIG compatibility requires that authentication methods
+        # use two factors.
+        await self.middleware.call('auth.set_authenticator_assurance_level', 'LEVEL_2')
+
+        # io_uring significantly complicates ability to use auditd to monitor file
+        # access and changes, and so we globally disable it when doing STIG
+        # compatibility.
+        await self.middleware.run_in_thread(set_io_uring_enabled, False)
+
+    @private
+    async def validate_stig(self, current_cred):
+        # The following validation steps ensure that users have the ability to
+        # manage the TrueNAS server after enabling STIG compatibility.
+        two_factor = await self.middleware.call('auth.twofactor.config')
+        if not two_factor['enabled']:
+            raise ValidationError(
+                'system_security_update.enable_gpos_stig',
+                'Two factor authentication must be globally enabled before '
```

Comment:
Might be worth it to make this two factor validation error a static string and re-use it since it's used in multiple places.